src/app/public/modules/back-to-top/back-to-top.directive.ts

```diff
@@ -89,12 +92,17 @@ export class SkyBackToTopDirective implements AfterViewInit, OnDestroy {
     private element: ElementRef
   ) {}
 
-  public ngAfterViewInit(): void {
-    const scrollableParent = this.domAdapter.findScrollableParent(this.element.nativeElement);
-    this.elementInView = this.domAdapter.isElementScrolledInView(this.element.nativeElement, scrollableParent);
+  public ngAfterViewChecked(): void {
```

Comment:
`ngAfterViewChecked` gets called quite frequently. I don't think this is the right choice.

Reply:
I 100% agree that it is called a lot, and I normally would avoid using it; however, in this case we need to ensure that the scrollable parent is updated when it changes. And since we don't have access to anything in the parent hierarchy we have to do that on change detection. This lifecycle hook is called when change detection fires and thus allows us to do that check. I don't like hooking into change detection but for this case - I couldn't find a way to avoid it.